spec/classes/init_spec.rb

```diff
@@ -456,369 +454,311 @@
-
-        it {
-          should contain_file('pam.d-service-testservice').with({
-            'ensure'  => 'file',
-            'path'    => '/etc/pam.d/testservice',
-            'owner'   => 'root',
-            'group'   => 'root',
-            'mode'    => '0644',
-          })
-        }
-
-        it { should contain_file('pam.d-service-testservice').with_content('foo') }
-      end
+      describe 'config files' do
+        context 'with default params' do
+          v[:files].each do |file|
+            group = file[:group] || 'root'
+            dirpath = file[:dirpath] || '/etc/pam.d/'
+
+            file[:types].each do |type|
+              filename = "#{file[:prefix]}#{type}#{file[:suffix]}"
+              path = "#{dirpath}#{file[:prefix]}#{type}#{file[:suffix]}"
+              path.gsub! '_', '-'
+              path.sub! 'pam-', ''
+              it do
+                should contain_file(filename).with({
+                  'ensure'  => 'file',
+                  'path'    => path,
+                  'owner'   => 'root',
+                  'group'   => group,
+                  'mode'    => '0644',
+                })
+              end
+              fixture = File.read(fixtures("#{filename}.defaults.#{k}"))
+              it { should contain_file(filename).with_content(fixture) }
 
-      context "with default params on #{v[:facts_hash][:osfamily]} with os.release.major #{v[:facts_hash][:os]}" do
-        let :facts do
-          v[:facts_hash]
-        end
+              v[:packages].sort.each do |pkg|
+                if v[:facts_hash][:osfamily] != 'Solaris' and (v[:facts_hash][:osfamily] != 'Suse' and v[:facts_hash][:os] != 9)
+                  it { should contain_file(filename).that_requires("Package[#{pkg}]") }
+                end
+              end
 
-        v[:files].each do |file|
-          group = file[:group] || 'root'
-          dirpath = file[:dirpath] || '/etc/pam.d/'
-
-          file[:types].each do |type|
-            filename = "#{file[:prefix]}#{type}#{file[:suffix]}"
-            path = "#{dirpath}#{file[:prefix]}#{type}#{file[:suffix]}"
-            path.gsub! '_', '-'
-            path.sub! 'pam-', ''
-            it {
-              should contain_file(filename).with({
-                'ensure'  => 'file',
-                'path'    => path,
-                'owner'   => 'root',
-                'group'   => group,
-                'mode'    => '0644',
-              })
-            }
-            fixture = File.read(fixtures("#{filename}.defaults.#{k}"))
-            it { should contain_file(filename).with_content(fixture) }
+              if file[:symlink]
+                symlinkname = "#{file[:prefix]}#{type}"
+                symlinkpath = "#{dirpath}#{file[:prefix]}#{type}"
+                symlinkpath.gsub! '_', '-'
+                symlinkpath.sub! 'pam-', ''
+                it do
+                  should contain_file(symlinkname).with({
+                    'ensure' => 'symlink',
```

Comment:
ensure would be `link` instead of `symlink`.

Reply:
Good find, I'll have to change it in the code as well.
https://github.com/ghoneycutt/puppet-module-pam/blob/puppet5/manifests/init.pp#L162